Userland/Libraries/LibWeb/HTML/HTMLTextAreaElement.cpp

```diff
@@ -96,6 +96,49 @@ void HTMLTextAreaElement::form_associated_element_was_removed(DOM::Node*)
     set_shadow_root(nullptr);
 }
 
+// https://html.spec.whatwg.org/multipage/form-elements.html#dom-textarea-defaultvalue
+String HTMLTextAreaElement::default_value() const
+{
+    // The defaultValue attribute's getter must return the element's child text content.
+    return child_text_content();
+}
+
+// https://html.spec.whatwg.org/multipage/form-elements.html#dom-textarea-defaultvalue
+void HTMLTextAreaElement::set_default_value(String const& default_value)
+{
+    // The defaultValue attribute's setter must string replace all with the given value within this element.
+    string_replace_all(default_value);
+}
+
+// https://html.spec.whatwg.org/multipage/form-elements.html#dom-textarea-value
+String HTMLTextAreaElement::value() const
+{
+    // The value IDL attribute must, on getting, return the element's API value.
+    return m_raw_value;
+}
+
+// https://html.spec.whatwg.org/multipage/form-elements.html#dom-output-value
+void HTMLTextAreaElement::set_value(String const& value)
+{
+    // FIXME: 1. Let oldAPIValue be this element's API value.
+
+    // 2. Set this element's raw value to the new value.
+    m_raw_value = value;
+
+    // 3. Set this element's dirty value flag to true.
+    m_dirty = true;
+
+    // FIXME: 4. If the new API value is different from oldAPIValue, then move the text entry cursor position to the end of the text control, unselecting any selected text and resetting the selection direction to "none".
+    update_placeholder_visibility();
+}
+
+// https://html.spec.whatwg.org/multipage/form-elements.html#dom-textarea-textlength
+unsigned long HTMLTextAreaElement::text_length() const
+{
+    // The textLength IDL attribute must return the length of the element's API value.
+    return m_raw_value.code_points().length();
```

Comment:
This is returning the number of code points, not the number of utf16 code units, see: https://infra.spec.whatwg.org/#string-length

Also I believe we generally prefer fixed width typed for this type of thing (we generate it in IDL as `u32`).

Reply:
Good point, I have fixed it